frontend/admin/src/js/components/user/GeneralInfoTabDialogs.tsx

```diff
@@ -0,0 +1,690 @@
+import React, { useState } from "react";
+import { useIntl } from "react-intl";
+import Dialog from "@common/components/Dialog";
+import Button from "@common/components/Button";
+import { getLocale } from "@common/helpers/localize";
+import { enumToOptions } from "@common/helpers/formUtils";
+import { getPoolCandidateStatus } from "@common/constants/localizedConstants";
+import { InputError, InputWrapper } from "@common/components/inputPartials";
+import { toast } from "react-toastify";
+import { UserRemoveIcon } from "@heroicons/react/solid";
+import { isEmpty } from "lodash";
+import {
+  CreatePoolCandidateAsAdminInput,
+  Pool,
+  PoolCandidate,
+  PoolCandidateStatus,
+  UpdatePoolCandidateAsAdminInput,
+  useCreatePoolCandidateMutation,
+  useDeletePoolCandidateMutation,
+  User,
+  useUpdatePoolCandidateMutation,
+} from "../../api/generated";
+
+export interface TableDialogProps {
+  selectedCandidate: PoolCandidate | null;
+  user: User;
+  onDismiss: () => void;
+}
+
+interface CloseDialogButtonProps {
+  close: () => void;
+}
+
+const CloseDialogButton: React.FC<CloseDialogButtonProps> = ({ close }) => {
+  const intl = useIntl();
+  return (
+    <Button type="button" mode="outline" color="secondary" onClick={close}>
+      <span data-h2-font-style="b(underline)">
+        {intl.formatMessage({
+          defaultMessage: "Cancel and go back",
+          description: "Close dialog button",
+        })}
+      </span>
+    </Button>
+  );
+};
+
+interface ConfirmDialogButtonProps {
+  onConfirm: () => void;
+  title: string;
+  icon?: React.FC<{ style: { width: string } }>;
+}
```

Comment:
I'm not sure if we are doing this elsewhere but it may be a good idea to disable the submit button while a mutation is running to prevent multiple submissions at once.